packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -302,7 +302,8 @@ class IOSDevice extends Device {
   }
 
   @override
-  bool isSupported() => true;
+  // 32-bit devices are not supported.
+  bool isSupported() => cpuArchitecture == DarwinArch.arm64;
```

Comment:
Does this never get called with `cpuArchitecture == x64` on simulators?

Reply:
No, this is in `IOSDevice` which is for physical devices.  Simulators are handled here:
https://github.com/flutter/flutter/blob/3109073fd9f9219d81ba20d5cf77ec12e708b53f/packages/flutter_tools/lib/src/ios/simulators.dart#L405